module/VuFind/src/VuFind/Controller/Feature/RecordVersionsSearchTrait.php

```diff
@@ -94,8 +90,8 @@ public function versionsAction()
         // won't in RSS mode):
         if (isset($view->results)) {
             $view->results->getUrlQuery()
-                ->setDefaultParameter('id', $id)
-                ->setDefaultParameter('keys', $keys)
+                ->setDefaultParameter('id', $this->params()->fromQuery('id'))
+                ->setDefaultParameter('keys', $this->params()->fromQuery('keys'))
```

Comment:
Is there a reason not to use `$driverAndKeys['keys']` here to avoid re-processing the user input?

It's also tempting to use `$record->getUniqueId()` in the line above, but I suspect that won't work since `$record` might be null (though I didn't trace the code super-thoroughly, so maybe the null case can/should avoid this logic). Should we include the original ID in the $driverAndKeys array, or is there another better way to handle this?

Reply:
I changed the helper method to include the id in the result and use it.